### PR TITLE
Complete refactoring of recipe configurations.

### PR DESCRIPTION
**Pull request by godofredoc**
Bug:
  https://github.com/flutter/flutter/issues/56419

### CODE REVIEW

File: config/iostools_config.star [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env lucicfg
+# Copyright 2020 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+"""
+Configurations for ios_tool builders.
+
+The schedulers pull commits indirectly from GoB repo (https://chromium.googlesource.com/external/github.com/flutter/flutter)
```
Comment (christopherfujino) [inline config/iostools_config.star:8]: commits actually come from https://flutter-mirrors.googlesource.com/

File: config/cocoon_config.star [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env lucicfg
+# Copyright 2019 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+"""
+Configurations for the cocoon repository.
+
+The schedulers pull commits indirectly from GoB repo (https://chromium.googlesource.com/external/github.com/flutter/flutter)
```
Comment (christopherfujino) [inline config/cocoon_config.star:8]: pulls from https://chromium.googlesource.com/external/github.com/flutter/cocoon

File: config/packaging_config.star [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        cipd_package='flutter/recipe_bundles/flutter.googlesource.com/recipes',
+        cipd_version='refs/heads/master',
+    )
+
+
+def packaging_prod_config(platform_args, branch, version, ref):
+
+    # Defines console views for prod builders
+    console_view_name = ('packaging'
+                         if branch == 'master' else '%s_packaging' % branch)
```
Comment (christopherfujino) [inline config/packaging_config.star:63]: this function should never get called with `branch == 'master'`, right?

Comment (christopherfujino) [reply config/packaging_config.star:63]: Should we throw here (or somehow escalate a warning in a way we'll see) if this function ever gets invoked on the master branch? I'm worried a future refactor will lead to us building packages for every single master commit. At the minimum, we should exit early.

Comment (godofredoc) [reply config/packaging_config.star:63]: Added an explanation and added functionality to fail if we ever try to build from master.

File: config/cocoon_config.star (hunk 2 above)
Comment (godofredoc) [reply config/cocoon_config.star:8]: Updated

File: config/iostools_config.star (hunk 1 above)
Comment (godofredoc) [reply config/iostools_config.star:8]: Updated.

**Comment by godofredoc** [comment]
> 2 nits about comments and then a question about ensuring we don't regress in the future and build packages on master.

Updated the PR addressing the comments.

**Pull request merged**